tempodb/encoding/vparquet/block_search.go

```diff
@@ -48,6 +48,7 @@ func (b *backendBlock) openForSearch(ctx context.Context, opts common.SearchOpti
 	if b.pf != nil && b.readerAt != nil {
 		// Reset metrics, is there a better way to do this?
 		b.readerAt.TotalBytesRead.Store(0)
+		// TODO: We should reset the context here, but it's not possible to do so without creating a new parquet.File.
```

Comment:
this is a nasty context leak.

when working on the wal i found that it was actually more memory efficient to _not_ cache parquet.files and the performance penalty of repeatedly opening was quite low. would recommend pushing a test image that does not do this file caching and see if it hurts perf.